server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -91,6 +91,7 @@ public void start() {
     }
 
     private void startWorkqueue() {
+        channelPool.start();
```

Comment:
I added that cause with the distributed task manager bindings, the sender was null in this case... But i'm not sure it's the right way to do

Reply:
You are rigth, that is the wrong way to fix this.

This is definitly a `Startable` intialization order. Because RabbitMQWorkQueue is not called by a Guice InitalizationOperation, then Guice have no clue that it should be started after the RabbitMQChannelPool.

The right way of doing this is to no longer call RabbitMQWorkQueue::start in RabbitMQWorkQueueSupplier

Instead, we should call it manually in tests, and rely on an initializationOperation within DistributedTaskManagerModule.

Reply:
Yes I tried a bit this way yesterday but I lacked time, I just wanted something quick to test... Probably shouldn't have !